app/src/main/java/com/lagradost/cloudstream3/ui/result/ResultTrailerPlayer.kt

```diff
@@ -172,12 +175,22 @@ open class ResultTrailerPlayer : ResultFragmentPhone(), IOnBackPressed {
         }
     }
 
-    override fun onBackPressed(): Boolean {
-        return if (isFullScreenPlayer) {
-            updateFullscreen(false)
-            false
-        } else {
-            true
+    private var backPressedCallback: OnBackPressedCallback? = null
+
+    private fun attachBackPressedCallback() {
+        if (backPressedCallback == null) {
+            backPressedCallback = object : OnBackPressedCallback(true) {
+                override fun handleOnBackPressed() {
+                    updateFullscreen(false)
+                }
+            }
         }
+
+        backPressedCallback?.isEnabled = true
+        requireActivity().onBackPressedDispatcher.addCallback(requireActivity(), backPressedCallback!!)
```

Comment:
Please refrain from ever using requireActivity() or !!

Even if it should logically never be an issue android has a habit of being extremely weird and if it can crash it will crash. Please use activity?. and other such null safe behavior.

Reply:
Sorry about that, fixed and thanks!